File: coding-interview/src/main/java/org/example/Game.java
package org.example;

import java.util.Random;

public class Game {
    private int playerAWins = 0;
    private int playerBWins = 0;
    private int draws = 0;

    private Random random = new Random();
    private Move getRandomMove() {
        int randomNumber = random.nextInt(3);
        Move randomMove = Move.values()[randomNumber];
        return randomMove;
    }
    private void playRound() {
        Move player_A_Move = getRandomMove();
        Move player_B_Move = Move.ROCK;

        Result result = getRoundResult(player_A_Move, player_B_Move);

        switch (result){
            case WIN:
                playerAWins++;
                break;
            case LOSE:
                playerBWins++;
                break;
            case DRAW:
                draws++;
        }
    }

    public Result getRoundResult(Move playerAMove, Move playerBMove){
        if(playerAMove.equals(Move.ROCK)){
            return Result.DRAW;
        } else if (playerAMove.equals(Move.PAPER)) {
            return Result.WIN;
        }else {
            return Result.LOSE;
        }
    }
    private void printResults(){
        System.out.println("Player A wins "+playerAWins +" of "+getTotalGames()+" games");
        System.out.println("Player B wins "+playerBWins+" of "+getTotalGames()+" games");
        System.out.println("Draws: "+draws+" of "+getTotalGames()+" games");
    }

    public int getTotalGames(){
        return playerAWins+playerBWins+draws;
    }

    public void playGame(int rounds){
        for (int i = 0; i < rounds; i++) {
            playRound();
        }
        printResults();
    }

//    public static void main(String[] args) {
//        for (int i = 0; i < 100; i++) {
//            playRound();
//        }
//        printResults();
//
//
//    }
}

File: coding-interview-kotline/src/main/kotlin/org/interview/coding/Game.kt
package org.interview.coding

class Game {
    private var playerAWins = 0
    private var playerBWins = 0
    private var draws = 0

    private fun getRandomMove(): Move {
        return Move.entries.toTypedArray().random()
    }

    private fun playRound(){
        val player_A_Move = getRandomMove()
        val player_B_Move = Move.ROCK

        val result = getRoundResult(player_A_Move, player_B_Move)
        when (result) {
            Result.WIN -> playerAWins++
            Result.LOSE -> playerBWins++
            Result.DRAW -> draws++
        }
    }
    fun getRoundResult(playerAMove: Move, playerBMove: Move): Result {

      return when{
            playerAMove == playerBMove -> Result.DRAW

            playerAMove == Move.ROCK && playerBMove == Move.SCISSORS ||
                    (playerAMove == Move.ROCK && playerBMove == Move.LIZARD) ||
                    (playerAMove == Move.PAPER && playerBMove == Move.ROCK)||
                    (playerAMove == Move.PAPER && playerBMove == Move.SPOCK)||
                    (playerAMove == Move.SCISSORS && playerBMove == Move.PAPER)||
                    (playerAMove == Move.SCISSORS && playerBMove == Move.LIZARD)||
                    (playerAMove == Move.LIZARD && playerBMove == Move.PAPER)||
                    (playerAMove == Move.LIZARD && playerBMove == Move.SPOCK) ||
                    (playerAMove == Move.SPOCK && playerBMove == Move.SCISSORS)||
                    (playerAMove == Move.SPOCK && playerBMove == Move.ROCK)
            -> Result.WIN
            else -> Result.LOSE
        }
    }

    private fun printResults() {
        println(("Player A wins " + playerAWins + " of " + getTotalGames()).toString() + " games")
        println(("Player B wins " + playerBWins + " of " + getTotalGames()).toString() + " games")
        println(("Draws: " + draws + " of " + getTotalGames()).toString() + " games")
    }

    fun getTotalGames(): Int {
        return playerAWins + playerBWins + draws
    }

    fun playGame(rounds: Int) {
        for (i in 0 until rounds) {
            playRound()
        }
        printResults()
    }
}
